src/aws_cloudwatch.go

```diff
@@ -202,7 +202,7 @@ func queryAvailableDimensions(resource string, namespace *string, clientCloudwat
 	return dimensions
 }
 
-func getDimensions(service *string, resourceArn *string, clientCloudwatch cloudwatchInterface) (dimensions []*cloudwatch.Dimension) {
+func getDimensions(service *string, resourceArn *string, configuredDimensions []dimension, clientCloudwatch cloudwatchInterface) (dimensions []*cloudwatch.Dimension) {
```

Comment:
I think we should rename getDimensions to detectDimensionsByService. That way getDimensions is only for auto dimension discovery and does not two things at once.